File: backend/NordicWebHub.Api/Models/MaintenanceLog.cs
namespace NordicWebHub.Api.Models;

public class MaintenanceLog
{
    public int Id { get; set; }

    public int CompanyId { get; set; }

    public Company Company { get; set; } = null!;

    public string Title { get; set; } = string.Empty;

    public string Description { get; set; } = string.Empty;

    public string ActionTaken { get; set; } = string.Empty;

    public string Result { get; set; } = string.Empty;

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/nordicwebhub-client/src/types/maintenanceLog.ts
export type MaintenanceLog = {
  id: number
  companyId: number
  companyName: string
  title: string
  description: string
  actionTaken: string
  result: string
  createdAt: string
}

export type CreateMaintenanceLogDto = {
  companyId: number
  title: string
  description: string
  actionTaken: string
  result: string
}

export type UpdateMaintenanceLogDto = CreateMaintenanceLogDto
